src/overlays/actors/ovl_En_Gg2/z_en_gg2.c

```diff
@@ -28,48 +39,456 @@ const ActorInit En_Gg2_InitVars = {
+            gSaveContext.weekEventReg[20] &= (u8)~8;
+            gSaveContext.weekEventReg[20] |= 0x10;
+        }
+
+        if (this->unk_1D8 != NULL) {
+            func_80B3B7E4(this->unk_1D8, this->unk_1DC, &this->actor.world.pos, &sp30);
+            Math_SmoothStepToS(&this->actor.world.rot.y, sp30.y, 4, 0x3E8, 1);
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            Math_SmoothStepToS(&this->actor.shape.rot.x, sp30.x, 4, 0x3E8, 1);
+            this->actor.world.rot.x = -this->actor.shape.rot.x;
+
+            if (func_80B3B648(this, this->unk_1D8, this->unk_1DC)) {
+                if (this->unk_1DC < (this->unk_1D8->count - 1)) {
+                    this->unk_1DC++;
+                } else {
+                    this->unk_2F1 = 1;
+                    if (globalCtx->sceneNum == SCENE_11GORONNOSATO) {
+                        gSaveContext.weekEventReg[20] |= 4;
+                        gSaveContext.weekEventReg[20] &= (u8)~8;
+                        gSaveContext.weekEventReg[20] &= (u8)~0x10;
+                    } else if (globalCtx->sceneNum == SCENE_17SETUGEN) {
+                        gSaveContext.weekEventReg[20] &= (u8)~4;
+                        gSaveContext.weekEventReg[20] |= 8;
+                        gSaveContext.weekEventReg[20] &= (u8)~0x10;
+                    } else if (globalCtx->sceneNum == SCENE_10YUKIYAMANOMURA) {
+                        gSaveContext.weekEventReg[20] &= (u8)~4;
+                        gSaveContext.weekEventReg[20] &= (u8)~8;
+                        gSaveContext.weekEventReg[20] |= 0x10;
+                    }
+                }
+            }
+        }
+    }
+    Math_ApproachF(&this->actor.speedXZ, 5.0f, 0.2f, 1.0f);
+}
+
+void func_80B3B4B0(EnGg2* this, GlobalContext* globalCtx) {
+    Vec3f sp64;
+    Color_RGBA8 sp60 = D_80B3C080;
+    Color_RGBA8 sp5C = D_80B3C084;
+    s32 i;
+
+    for (i = 0; i < 5; i++) {
+        sp64.x = (Rand_Centered() * 50.0f) + this->unk_304.x;
+        sp64.y = this->unk_304.y;
+        sp64.z = (Rand_Centered() * 50.0f) + this->unk_304.z;
+        EffectSsKiraKira_SpawnDispersed(globalCtx, &sp64, &D_80B3C088, &D_80B3C094, &sp60, &sp5C, 3000, 40);
+    }
+}
+
+void func_80B3B5D4(EnGg2* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->unk_2E4)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->unk_2E4, &this->actor);
+        this->actionFunc = func_80B3AE60;
+    } else {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+        }
+        ActorCutscene_SetIntentToPlay(this->unk_2E4);
+    }
+}
+
+s32 func_80B3B648(EnGg2* this, Path* path, s32 arg2_) {
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    s32 sp58 = path->count;
+    s32 arg2 = arg2_;
+    s32 ret = false;
+    f32 phi_f12;
+    f32 phi_f14;
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    Vec3f sp30;
+
+    Math_Vec3s_ToVec3f(&sp30, &points[arg2]);
```

Comment:
Interesting that it wants a temp for the function argument `arg2` like this...

Reply:
All of these path functions are like this, yeah.